TimberAPI/SpecificationSystem/Fixes/CustomSpecifications/Golems/GolemFactionSpecificationObjectDeserializer.cs

```diff
@@ -0,0 +1,18 @@
+using System;
+using Timberborn.Persistence;
+
+namespace TimberbornAPI.SpecificationSystem.Fixes.CustomSpecifications.Golems
+{
+    public class GolemFactionSpecificationObjectDeserializer : IObjectSerializer<GolemFactionSpecification>
+    {
+        public void Serialize(GolemFactionSpecification value, IObjectSaver objectSaver)
+        {
+            throw new NotSupportedException();
+        }
+
+        public Obsoletable<GolemFactionSpecification> Deserialize(IObjectLoader objectLoader)
+        {
+            return (Obsoletable<GolemFactionSpecification>) new GolemFactionSpecification(objectLoader.Get(new PropertyKey<string>("FactionId")), objectLoader.Get(new PropertyKey<string>("GolemId")));
```

Comment:
The magic strings here should stored in a private fields instead

Reply:
Sure iirc this was basicly a copy paste of their code of another specification xD

Reply:
:D thats weird, considering they have propertykeys set up nicely in a lot of places